client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => (boxCounts[line.box]++ % BOX_PAGE_SIZE) < BOX_PAGE_SIZE)
```

Comment:
I think it should add stuff to a box that the user is looking at. I don't care as much about whether we have a large number of icons on the homescreen.

One way to do it would be

```js
.each(line => {
  if (this.boxesById[line.box].contents.length % BOX_PAGE_SIZE) {
    this.boxesById[line.box].contents.push(line);
  }
})
```

...which is slightly inefficient because it needs to check every box rather than slicing, but I kind of doubt the difference is noticeable for a few hundred uploads.